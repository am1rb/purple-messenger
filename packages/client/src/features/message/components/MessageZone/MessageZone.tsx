import React from 'react';
import clsx from 'clsx';
import useStyles from './styles';

interface Props {
  className: string;
  conversationId?: number;
}

function MessageZone({ className, conversationId }: Props) {
  const classes = useStyles();
  return (
    <div className={clsx(classes.root, className)}>
      {conversationId}
      please select a conversation
    </div>
  );
}

export default MessageZone;
