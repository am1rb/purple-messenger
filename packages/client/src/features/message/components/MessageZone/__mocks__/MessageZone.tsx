import React from "react";

interface MessageZoneProps {
  friendUsername: string | undefined;
}

const messageZone = jest.fn(({ friendUsername }: MessageZoneProps) => (
  <div data-testid="message-zone" data-username={friendUsername} />
));

export default messageZone;
