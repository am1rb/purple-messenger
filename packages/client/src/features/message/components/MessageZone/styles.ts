import { makeStyles } from '@material-ui/styles';
import { Theme } from '@material-ui/core';

export default makeStyles((theme: Theme) => ({
  root: {}
}));
