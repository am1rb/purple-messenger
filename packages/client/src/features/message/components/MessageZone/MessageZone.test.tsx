import React from "react";
import { render } from "@testing-library/react";
import MessageZone, { MessageZoneProps } from "./MessageZone";

jest.mock("../SendMessageForm");
jest.mock("../MessageList");

const sharedProps: MessageZoneProps = {
  username: "john",
};

describe("The <MessageZone /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<MessageZone {...sharedProps} />);
    expect(container).toMatchSnapshot();
  });

  it("Should disable the send message form prop the send form if there is no username", () => {
    const { getByTestId } = render(<MessageZone username={undefined} />);
    expect(getByTestId("mock-send-message-form").dataset).toMatchObject({
      disabled: "true",
    });
  });

  it("Should not disable the send message form if there is no username", () => {
    const { getByTestId } = render(<MessageZone {...sharedProps} />);
    expect(getByTestId("mock-send-message-form").dataset).toMatchObject({
      disabled: "false",
    });
  });
});
