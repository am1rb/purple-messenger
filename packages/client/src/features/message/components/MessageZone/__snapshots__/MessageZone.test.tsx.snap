// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <MessageZone /> tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mui-box"
  >
    <div
      data-testid="mui-box"
    >
      <div
        data-testid="mock-message-list"
      />
    </div>
    <div
      data-testid="mui-box"
    >
      <form
        data-disabled="false"
        data-testid="mock-send-message-form"
      />
    </div>
  </div>
</div>
`;
